# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/sdn/sdnAccessible.cpp
# Commit: a53fb045a8e3
# Full Hash: a53fb045a8e34c1e1f14ca7b705eb9a5a0e76db2
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-05-02 09:29:52
# Regressor Bug: 1694865
# File Overlap Count: 1
# Description:
#   Bug 1694865 part 19: Make EnumVariant, GeckoCustom and sdn*Accessible classes aggregate MsaaAccessible/MsaaDocAccessible instead of AccessibleWrap/DocAccessibleWrap. r=morgan
#   
#   These objects need to aggregate the primary COM object (MsaaAccessible).
#   Once the IUnknown implementation moves out of AccessibleWrap, it won't be possible to aggregate *AccessibleWrap any more.
#   
# ==============================================================================

diff -r 5bd374b642fe -r a53fb045a8e3 accessible/windows/sdn/sdnAccessible.cpp
--- a/accessible/windows/sdn/sdnAccessible.cpp	Sat May 01 21:09:04 2021 +0000
+++ b/accessible/windows/sdn/sdnAccessible.cpp	Sat May 01 22:29:25 2021 +0000
@@ -52,7 +52,7 @@
     return S_OK;
   }
 
-  AccessibleWrap* accessible = GetAccessible();
+  MsaaAccessible* accessible = GetMsaa();
   if (accessible) return accessible->QueryInterface(aREFIID, aInstancePtr);
 
   // IUnknown* is the canonical one if and only if this accessible doesn't have
@@ -105,9 +105,9 @@
   // application can compare this to the childID we return for events such as
   // focus events, to correlate back to data nodes in their internal object
   // model.
-  AccessibleWrap* accessible = GetAccessible();
+  MsaaAccessible* accessible = GetMsaa();
   if (accessible) {
-    *aUniqueID = MsaaAccessible::GetChildIDFor(accessible);
+    *aUniqueID = MsaaAccessible::GetChildIDFor(accessible->LocalAcc());
   } else {
     if (mUniqueId.isNothing()) {
       MsaaAccessible::AssignChildIDTo(WrapNotNull(this));